src/Workspaces/CoreTest/FindAllDeclarationsTests.cs

```diff
@@ -689,6 +692,8 @@ public async Task TestSymbolTreeInfoSerialization()
 reader, Checksum.Null);
 
             info.AssertEquivalentTo(readInfo);
+
+            GC.KeepAlive(keepAliveObjects);
```

Comment:
same comments.  would like these tests to assert these are empty.

Reply:
Empty and not empty are both valid here.

Reply:
can you calrify that?  SymbolTreeInfo shouldn't be causing any objects to be kept alive.

Reply:
SymbolTreeInfo is allowed to keep objects alive as part of serialization/deserialization, particularly if it can leverage that feature to improve performance. The aggregate cost of serialization/deserialization in tests is very high, so we provide the full set of available efficiency features for tests to use if/when needed. Even if this type does not use the feature today, it is allowed to start doing so at any time for any reason and the test remains valid.

Reply:
➡️ Added trip-wire assertions with messages suggesting that the assertion is part of change detection as opposed to being part of behavior validation.